tests/unit/anchore_engine/services/policy_engine/engine/feeds/test_sync_utils.py

```diff
@@ -0,0 +1,271 @@
+                "grype",
+                [
+                    FeedConfiguration("vulnerabilities", False),
+                    FeedConfiguration("nvdv2", False),
+                    FeedConfiguration("grypedb", True),
+                    FeedConfiguration("packages", True),
+                ],
+                ["grypedb", "packages"],
+                ["grypedb"],
+            ),
+            (  # legacy provider with packages and grypedb enabled
+                "legacy",
+                [
+                    FeedConfiguration("vulnerabilities", False),
+                    FeedConfiguration("nvdv2", False),
+                    FeedConfiguration("grypedb", True),
+                    FeedConfiguration("packages", True),
+                ],
+                ["packages"],
+                ["packages"],
+            ),
+        ],
+    )
+    def test_config_filtering(
+        self,
+        provider: str,
+        feed_configurations: List[FeedConfiguration],
+        expected_to_sync_after_compute: List[str],
+        expected_to_sync_after_filtering: List[str],
+    ):
+        """
+        This is a bit confusing and probably should be changed, which is why i've written a test for it.
+        There are two SyncUtilProviders.
+        The LegacySyncUtilProvider works for all feeds that follow the legacy format.
+        The GrypeDBSyncUtilProvider works for the GrypeDB feed format.
+        However, the VulnerabilitiesProvider has two implementations.
+        The LegacyProvider contains all vulnerability logic that changes when the provider is set to "legacy"
+        The GrypeProvider contains all vulnerability logic that changes when the provider is set to "grype"
+        As such, the GrypeProvider actually returns both "packages" and "grypedb" SyncConfigs,
+        while "packages" is actually a Legacy style feed.
+        Meanwhile, the "packages" feed can only be synced by the LegacySyncUtilProvider.
+        The solution is likely to wrap the entire sync method with the SyncUtilProvider, that way LegacySyncUtilProvider
+        can just do legacy feeds, while GrypeDBSyncUtilProvider will first do "grypedb" feed with the grype logic
+        and then do "packages" feed with the legacy logic.
+        """
+        if provider == "legacy":
+            vulnerabilities_provider = LegacyProvider()
+        else:
+            vulnerabilities_provider = GrypeProvider()
+        sync_configs = compute_selected_configs_to_sync(
+            provider=vulnerabilities_provider.get_config_name(),
+            vulnerabilities_config=get_config_for_params(provider, feed_configurations),
+            default_provider_sync_config=vulnerabilities_provider.get_default_sync_config(),
+        )
+        assert set(sync_configs.keys()) == set(expected_to_sync_after_compute)
+        sync_utils_provider = vulnerabilities_provider.get_sync_utils(sync_configs)
+        filtered_configs = sync_utils_provider._get_filtered_sync_configs(sync_configs)
```

Comment:
This should probably be 2 different tests since it is testing 2 different cases and function outputs.

Reply:
Yeah you're right. I sort of wanted to keep it like this because this test was also supposed to indirectly serve as documentation by sort of mirroring the order of ops that happen when the sync starts. I recognize that's not really ideal. I'll split it up and leave the big docstring in.